File: admin-mobile/scripts/patch_background.js
const fs = require('fs');
const path = require('path');

const appDir = path.join(__dirname, '../src/app/(app)');
const files = fs.readdirSync(appDir);

files.forEach(file => {
    if (file.endsWith('.tsx')) {
        const filePath = path.join(appDir, file);
        let content = fs.readFileSync(filePath, 'utf8');

        if (content.includes('LinearGradient') && content.includes('expo-linear-gradient')) {
            if (!content.includes('AppBackground')) {
                content = content.replace(
                    /(import\s+\{[^}]*\}\s+from\s+['"]expo-linear-gradient['"];?\n?)/g,
                    "$1import AppBackground from '../../components/AppBackground';\n"
                );
            }

            content = content.replace(
                /<LinearGradient\s+colors=\{\[colors\.gradientStart,\s*colors\.gradientEnd,\s*colors\.gradientStart\]\}/g,
                "<AppBackground"
            );
            
            // For profile.tsx loading state or others
            content = content.replace(
                /<LinearGradient\s+colors=\{\[colors\.gradientStart,\s*colors\.gradientEnd\]\}/g,
                "<AppBackground"
            );
            
            content = content.replace(/<\/LinearGradient>/g, "</AppBackground>");

            fs.writeFileSync(filePath, content, 'utf8');
        }
    }
});

console.log("Done replacing LinearGradient.");

File: admin-mobile/scripts/patch_background.py
import os
import re

app_dir = r"C:\Users\PRINTARA\Documents\GitHub\shop-co\admin-mobile\src\app\(app)"

for filename in os.listdir(app_dir):
    if filename.endswith(".tsx"):
        filepath = os.path.join(app_dir, filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        if "LinearGradient" in content and "expo-linear-gradient" in content:
            # Add AppBackground import if missing
            if "AppBackground" not in content:
                content = re.sub(
                    r"(import\s+\{[^}]*\}\s+from\s+['\"]expo-linear-gradient['\"];?\n?)",
                    r"\1import AppBackground from '../../components/AppBackground';\n",
                    content
                )
            
            # Replace <LinearGradient colors={[colors.gradientStart, colors.gradientEnd, colors.gradientStart]}
            content = re.sub(
                r"<LinearGradient\s+colors=\{\[colors\.gradientStart,\s*colors\.gradientEnd,\s*colors\.gradientStart\]\}",
                r"<AppBackground",
                content
            )
            # Replace </LinearGradient>
            content = content.replace("</LinearGradient>", "</AppBackground>")
            
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
                
print("Done replacing LinearGradient.")
